caffe2/serialize/inline_container.h

```diff
@@ -59,195 +63,271 @@ namespace torch { namespace jit {
 //         been written. We place the variable-length index at the end and do
 //         not put any indicies into the header to fulfill this constraint.
 
+// The serialized model, which contains all the metadata information,
```

Comment:
I guess it's more because of the continuous writing. If header is in front I might not yet know beforehand what are the offsets of all tensors are because the serialized header depends on them. That's why putting header in the end allows to hard code offsets. That's the format of ZIP file too for example. (and allows for appending more tensors easily though we probably care less of this one)